src/redisai.c

```diff
@@ -92,19 +92,27 @@ typedef struct RunQueueInfo {
   pthread_mutex_t run_queue_mutex;
   pthread_cond_t queue_condition_var;
   queue *run_queue;
-  pthread_t run_thread_id;
+  pthread_t *threads;
 } RunQueueInfo;
 
+static AI_dict *run_queues = NULL;
+static int perqueueThreadPoolSize = 1;
+
 void freeRunQueueInfo(RunQueueInfo* info) {
   if (info->run_queue) {
     RedisModule_Free(info->run_queue);
   }
-
+  if (info->threads){
+    /* Wait for workers to exit */
+    for (int i = 0; i < perqueueThreadPoolSize; i++){
+      const int rtn = pthread_join(info->threads[i], NULL);
```

Comment:
We should do something with `rtn`

Reply:
@lantiga would it be enough for now to return REDISMODULE_ERR? please revise latest commit.